rosetta-testing/src/test/java/com/regnosys/rosetta/types/SubtypeRelationTest.xtend

```diff
@@ -28,6 +36,166 @@ class SubtypeRelationTest {
 		return model.elements.filter(RosettaNamed).findFirst[it.name == name] as RosettaEnumeration
 	}
 	
+	@Test
+	def testJoinOnMetaSubTypesReturnsParent() {
+		val model = '''
+			type A:
+			
+			type B extends A:
+			
+			type C extends A:
+				
+		'''.parseRosetta
+		
+		val fieldB = '''
+			fieldB B (1..1)
+				[metadata reference]
+		'''.parseAttribute(#[model]).RTypeOfSymbol
+		
+		val fieldC = '''
+			fieldC C (1..1)
+				[metadata reference]
+		'''.parseAttribute(#[model]).RTypeOfSymbol
+		
+		
+		val fieldA = model.getData("A").buildRDataType
+		
+		val joined = fieldB.join(fieldC)
+		
+		assertEquals(joined.RType, fieldA)
+		assertFalse(joined.hasMeta)
```

Comment:
Might as well do `assertEquals(joined, fieldA.withNoMeta)`.

Also: "expected" and "actual" are wrong way around?

Other idea: just test on stringified version of type with meta you are testing.

Reply:
Same comment for other tests